### ISSUE TITLE
Expose Vagrantfile validation code to CLI

### ISSUE THREAD

**Issue by mschuchard** [mitchellh/vagrant#8151]
Are there any plans or some kind of hidden ability to expose the Vagrantfile validation code to come kind of CLI option? e.g. Terraform has validate and plan options in its CLI. I am interested in something analogous for Vagrant.

(If this sounds interesting, please mark as Feature Request; if not, go ahead and close and I can look at developing some kind of plugin for it)

**Comment by kikitux** [comment]
Hello.

Can you elaborate more?

What kind of errors you are aiming to find with this?

You could aim to build a plugin first, and see how things go

**Comment by mschuchard** [comment]
Say I have some trivial Vagrantfile:

```
Vagrant.configure('2') do |config|
  cnfig.vm.box = 'centos/7'
end
```

I will not know I have made a mistake until I perform a `vagrant` command. This is really annoying for me. If the validation code was exposed to the CLI, then people could easily develop tools for vim, Atom, Sublime, etc. to instantly know when mistakes are made. Also, having a CLI to validate Vagrantfiles could be useful in other scenarios such as CI.

**Comment by hasyimibhar** [comment]
There is a builtin action called `ConfigValidate` which is used internally by Vagrant to validate the config file. Maybe you can build a plugin that simply calls `ConfigValidate`.

**Comment by mschuchard** [comment]
@hasyimibhar Awesome, that was exactly what I was hoping for. I saw `ConfigValidate` in the source code and was looking for a way to access it with the CLI. Thank you for exposing it to the CLI. Subscribed to the PR.

**Comment by chrisroberts** [comment]
Resolved by #8264
